proxy/http3/Http3Session.cc

```diff
@@ -38,9 +38,8 @@ HQSession::HQSession(NetVConnection *vc) : ProxySession(vc)
 
 HQSession::~HQSession()
 {
-  for (HQTransaction *t = this->_transaction_list.head; t; t = static_cast<HQTransaction *>(t->link.next)) {
-    delete t;
-  }
+  // Transactions should be deleted first before HQSesson gets deleted.
+  ink_assert(this->_transaction_list.head);
```

Comment:
We originally had cleanup code here, but I realized that deleting a transaction modifies the transaction list. I removed the cleanup code since it was just a safety net.